Zend/zend_language_parser.y

```diff
@@ -548,6 +548,10 @@ function_declaration_statement:
 	backup_fn_flags '{' inner_statement_list '}' backup_fn_flags
 		{ $$ = zend_ast_create_decl(ZEND_AST_FUNC_DECL, $2 | $13, $1, $4,
 		      zend_ast_get_str($3), $6, NULL, $11, $8, NULL); CG(extra_fn_flags) = $9; }
+   | function returns_ref T_STRING backup_doc_comment '(' parameter_list ')' return_type
+        backup_fn_flags T_DOUBLE_ARROW expr ';' backup_fn_flags
+     		{ $$ = zend_ast_create_decl(ZEND_AST_FUNC_DECL, $2, $1, $4,
+     		      zend_ast_get_str($3), $6, NULL, zend_ast_create(ZEND_AST_RETURN, $11), $8, NULL); CG(extra_fn_flags) = $9; }
```

Comment:
nit: I think this would be more readable if you put the two cases inside of a `function_body` rule, like `method_body` is currently doing.

```c
function_body:
	 	'{' inner_statement_list '}'	{ $$ = $2; }
	|   T_DOUBLE_ARROW expr ';'         { $$ = zend_ast_create(ZEND_AST_RETURN, $2); }
;
```

Reply:
Joe W was able to help me factor that out, and then suggested also making method_body just delegate to function_body, which I did as well.  I also rebased while I was at it, because the code was from September. :-)